### ISSUE TITLE
Saving relation fields with changesOnly=true

### ISSUE THREAD

**Issue by aikoven** [js-data/js-data#76]
I have a resource with `hasMany` relation:
```
post = {
    id: 1,
    comments: []
}
```
When I add new entry to `comments` field and save `post` with `changesOnly: true`, all relation fields are ignored, thus there is nothing to save and adapter is bypassed. Although after that `post` is re-injected which gets new `comment` injected as well resulting in error saying that `comment` does not have `id` attribute.

**Comment by jmdobry** [comment]
This is because calling `DS#changes` for an item ignores changes inside of that items relations. Even if an item's relation has change, the item has not changed. Now, it may be that I should make this configurable, so you can choose to have changes in an item's relation cause the item to have changes, but I don't know about that.

As an alternative you could do this several other ways:

```js
Post.update(post.id, { comments: post.comments }).then(function (post) {
  post.comments[0].id; // has an id!
});
```

Or

```js
$scope.createComment = function () {
  Comment.create({
    content: $scope.newCommentText,
    post_id: $scope.post.id,
    user_id: $rootScope.loggedInUser.id
  });
};

Comment.bindAll({ post_id: $scope.post.id }, $scope, 'comments');
```

Or

```js
$scope.createComment = function () {
  var newComment = Comment.createInstance({
    content: $scope.newCommentText,
    post_id: $scope.post.id,
    user_id: $rootScope.loggedInUser.id
  });
  
  newComment.DSCreate().then(function (comment) {
    $scope.post.comments.push(comment);
  });
};
```

Lot's of ways to do it.

**Comment by aikoven** [comment]
But how do I actually save changes in `post`? I know that relation fields are ignored by `changesOnly` and I'm okay with that, I just want to send changes in plain fields to the server, although I get error because we try to inject `comment`s that were not yet saved. Is there a way to achieve that other than collecting changes by hand or deleting `post.comments` field?

**Comment by jmdobry** [comment]
Hmm, the store was not made to handle the injection of data that is missing a primary key. So, one thing that a lot of users do is add temporary primary keys to items that have not yet been saved so that they can be injected into the store. If you put temporary ids on your unsaved comments then the injection error should go away.

Manually collecting changes is definitely another approach (so you can use `update` instead of `save`), one that I myself have used in my own apps.

Let me think about this more. Perhaps there is something js-data can do better here to make things easier when saving items that are connected to relations. I have an open issue (#11) on this. I've kind of been gathering feedback and waiting for people to ask about it (show me their use cases) before I pursue any solution in the library.

**Comment by jmdobry** [comment]
I believe this will be fixed in v1.5.9
